drivers/peci/peci_npcx.c

```diff
@@ -0,0 +1,279 @@
+		 * The read block FCS is stored at offset N.
+		 */
+		peci_rx_buf->buf[i] = reg->PECI_RD_FCS;
+		LOG_DBG("Wr FCS:0x%02x|Rd FCS:0x%02x", reg->PECI_WR_FCS, reg->PECI_RD_FCS);
+	}
+
+out:
+	k_sem_give(&data->lock);
+	return ret;
+}
+
+static void peci_npcx_isr(const struct device *dev)
+{
+	const struct peci_npcx_config *const config = dev->config;
+	struct peci_npcx_data *const data = dev->data;
+	struct peci_reg *const reg = config->base;
+	uint8_t status;
+
+	status = reg->PECI_CTL_STS;
+	LOG_DBG("PECI ISR status: 0x%02x", status);
+	/*
+	 * Disable the transaction done interrupt, also clear the status bits
+	 * if they were set.
+	 */
+	reg->PECI_CTL_STS &= ~BIT(NPCX_PECI_CTL_STS_DONE_EN);
+
+	if (IS_BIT_SET(status, NPCX_PECI_CTL_STS_ABRT_ERR)) {
+		data->trans_error = NPCX_PECI_WR_ABORT_ERROR;
+		LOG_ERR("PECI Nego or Wr FCS(0x%02x) error", reg->PECI_WR_FCS);
+	} else if (IS_BIT_SET(status, NPCX_PECI_CTL_STS_CRC_ERR)) {
+		data->trans_error = NPCX_PECI_RD_CRC_ERROR;
+		LOG_ERR("PECI Rd FCS(0x%02x) error", reg->PECI_WR_FCS);
+	} else {
+		data->trans_error = NPCX_PECI_NO_ERROR;
+	}
+
+	k_sem_give(&data->trans_sync_sem);
+}
+
+static const struct peci_driver_api peci_npcx_driver_api = {
+	.config = peci_npcx_configure,
+	.enable = peci_npcx_enable,
+	.disable = peci_npcx_disable,
+	.transfer = peci_npcx_transfer,
+};
+
+static int peci_npcx_init(const struct device *dev)
+{
+	const struct device *const clk_dev = DEVICE_DT_GET(NPCX_CLK_CTRL_NODE);
+	const struct peci_npcx_config *const config = dev->config;
+	struct peci_npcx_data *const data = dev->data;
+	int ret;
+
+	if (!device_is_ready(clk_dev)) {
+		LOG_ERR("%s device not ready", clk_dev->name);
+		return -ENODEV;
+	}
+
+	ret = clock_control_on(clk_dev, (clock_control_subsys_t *)&config->clk_cfg);
+	if (ret < 0) {
+		LOG_ERR("Turn on PECI clock fail %d", ret);
+		return ret;
+	}
+
+	ret = clock_control_get_rate(clk_dev, (clock_control_subsys_t *)&config->clk_cfg,
+				     &data->peci_src_clk_freq);
+	if (ret < 0) {
+		LOG_ERR("Get PECI source clock rate error %d", ret);
+		return ret;
+	}
+
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+
+	if (ret != 0) {
+		LOG_ERR("NPCX PECI pinctrl init failed (%d)", ret);
+		return ret;
+	}
+
+	k_sem_init(&data->trans_sync_sem, 0, 1);
+	k_sem_init(&data->lock, 1, 1);
+
+	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQ(0, priority), peci_npcx_isr, DEVICE_DT_INST_GET(0),
+		    0);
+
+	return 0;
+}
+
+static struct peci_npcx_data peci_npcx_data0;
+
+PINCTRL_DT_INST_DEFINE(0);
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "only one 'nuvoton_npcx_peci' compatible node can be supported");
+
+static const struct peci_npcx_config peci_npcx_config0 = {
+	.base = (struct peci_reg *)DT_INST_REG_ADDR(0),
+	.clk_cfg = NPCX_DT_CLK_CFG_ITEM(0),
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
+};
+
+DEVICE_DT_INST_DEFINE(0, &peci_npcx_init, NULL, &peci_npcx_data0, &peci_npcx_config0, POST_KERNEL,
```

Comment:
This needs a check that only one nuvoton,npcx-peci node is defined.

Reply:
Done.